libs/azure-storage/tests/utils.py

```diff
@@ -69,18 +86,64 @@ def get_expected_documents(
 def get_test_blobs(
     blob_names: Optional[Union[str, Iterable[str]]] = None, prefix: Optional[str] = None
 ) -> list[dict[str, str]]:
+    return _get_test_blobs(_TEST_BLOBS, blob_names, prefix)
+
+
+def get_datalake_test_blobs(
+    blob_names: Optional[Union[str, Iterable[str]]] = None,
+    prefix: Optional[str] = None,
+    include_directories: Optional[bool] = False,
+) -> list[dict[str, Any]]:
+    if not include_directories:
+        return get_test_blobs(blob_names, prefix)
+
+    blobs = _get_test_blobs(_TEST_DATALAKE_BLOBS, blob_names, prefix)
+    return [
+        {
+            **blob,
+            "size": len(blob["blob_content"]),
+            "metadata": blob.get("metadata", None),
+        }
+        for blob in blobs
+    ]
```

Comment:
For the `size` and `metadata`, we should just lower this into our `_get_test_blobs()` helper function so that we are consistent on what data is returned. In general, size and metadata will be returned no matter the account type.